File: 2231.Largest_Number_After_Digit_Swaps_by_Parity.ts
const largestInteger = (num: number): number => {
    let num_str = num.toString();
    while (true) {
        let is = true;
        for (let i = num_str.length - 1; i >= 0; i--) {
            for (let j = 0; j < i; j++) {
                if (num_str[j] < num_str[i] && (parseInt(num_str[j]) % 2 === parseInt(num_str[i]) % 2)) {
                    num_str = num_str.substring(0, j) + num_str[i] + num_str.substring(j + 1, i) + num_str[j] + num_str.substring(i + 1, num_str.length);
                    is = false;
                }
            }
        }
        if (is) break;
    }
    return parseInt(num_str);
};

console.log(largestInteger(1234))

File: 2231.Largest_Number_After_Digit_Swaps_by_Parity.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int largestInteger(int num)
    {
        string num_str = to_string(num);

        while (true)
        {
            bool is = true;
            for (int i = num_str.length() - 1; i >= 0; i--)
            {
                for (int j = 0; j < i; j++)
                {
                    if (num_str[j] < num_str[i] && (num_str[i]) % 2 == (num_str[j] % 2))
                    {
                        swap(num_str[j], num_str[i]);
                        is = false;
                        break;
                    }
                }
            }
            if (is)
            {
                break;
            }
        }
        return stoi(num_str);
    }
};

int main()
{
}
// by Shihab Mahamud (github.com/shh26b)
// Monday, April 11, 2022 | 02:30:21 PM (+06)
